src/main/scala/com/intenthq/action_processor/integrations/repositories/MapDBRepository.scala

```diff
@@ -0,0 +1,31 @@
+package com.intenthq.action_processor.integrations.repositories
+
+import java.io.File
+import java.nio.file.Paths
+import java.time.LocalDateTime
+
+import cats.effect.{IO, Resource}
+import com.intenthq.action_processor.integrations.config.MapDbSettings
+import org.mapdb.{DB, DBMaker}
+
+object MapDBRepository {
+  def load(mapDBSettings: MapDbSettings): Resource[IO, DB] = {
+    val dbCreated = for {
+      now <- IO.delay(LocalDateTime.now())
+      dbFile <- IO.delay(new File(Paths.get(mapDBSettings.dbPath.toString, s"dbb-${now.toLocalDate}-${now.toLocalTime}").toUri))
+      _ <- IO.delay(dbFile.deleteOnExit())
+      createDb <- IO.delay {
+        DBMaker
+          .fileDB(dbFile.getAbsolutePath)
+          .allocateStartSize(mapDBSettings.startDbSize)
+          .allocateIncrement(mapDBSettings.incSize)
+          .fileMmapEnableIfSupported()
+          .fileMmapPreclearDisable()
+          .closeOnJvmShutdown()
+          .fileDeleteAfterClose()
+          .make()
+      }
+    } yield createDb
+    Resource.make(dbCreated)(db => IO.delay(db.close()))
```

Comment:
As far as I know this operation is blocking.
Also, in case of high `startDbSize` it usually takes long time.
I think we should explore the possibility of blocking it on Blocker pool.

Could we explore the possibility of passing it a Blocking pool?
```
def load(blocker: Blocker)(mapDBSettings: MapDbSettings)
```

```
Resource.fromAutoCloseableBlocking(blocker)(dbCreated)
```
or
```
Resource.make(blocker.blockOn(dbCreated))(db => IO.delay(db.close()))
```

Reply:
Changed as discussed